app/javascript/components/action/action-button.jsx

```diff
@@ -4,15 +4,28 @@ import style from './style.module.less';
 
 export const ActionButton = ({id, type, title, setActionItems}) => {
   const onClickHandler = async () => {
+    let actionItem;
     switch (type) {
       case 'close':
-        setActionItems(await actionItemsApi.closeActionItem(id));
+        actionItem = await actionItemsApi.closeActionItem(id);
+        if (actionItem) {
+          setActionItems(actionItem);
+        }
+
         break;
       case 'complete':
-        setActionItems(await actionItemsApi.completeActionItem(id));
+        actionItem = await actionItemsApi.completeActionItem(id);
+        if (actionItem) {
+          setActionItems(actionItem);
+        }
+
         break;
       case 'reopen':
-        setActionItems(await actionItemsApi.reopenActionItem(id));
+        actionItem = await actionItemsApi.reopenActionItem(id);
+        if (actionItem) {
```

Comment:
Maybe you can move `if (actionItem) { setActionItems(actionItem); }` below `switch` block?
As you set action items in each case